google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/JsonToProtoMessageTest.java

```diff
@@ -1019,6 +1019,128 @@ public void testRequired() throws Exception {
+                        TableFieldSchema.FieldElementType.newBuilder()
+                            .setType(TableFieldSchema.Type.DATE)
+                            .build())
+                    .setMode(TableFieldSchema.Mode.NULLABLE)
+                    .build())
+            .addFields(
+                TableFieldSchema.newBuilder()
+                    .setName("range_datetime_mixed_case")
+                    .setType(TableFieldSchema.Type.RANGE)
+                    .setRangeElementType(
+                        TableFieldSchema.FieldElementType.newBuilder()
+                            .setType(TableFieldSchema.Type.DATETIME)
+                            .build())
+                    .setMode(TableFieldSchema.Mode.NULLABLE)
+                    .build())
+            .addFields(
+                TableFieldSchema.newBuilder()
+                    .setName("range_timestamp_mixed_case")
+                    .setType(TableFieldSchema.Type.RANGE)
+                    .setRangeElementType(
+                        TableFieldSchema.FieldElementType.newBuilder()
+                            .setType(TableFieldSchema.Type.TIMESTAMP)
+                            .build())
+                    .setMode(TableFieldSchema.Mode.NULLABLE)
+                    .build())
+            .build();
+
+    TestRange expectedProto =
+        TestRange.newBuilder()
+            .setRangeDate(TestRangeDate.newBuilder().setStart(18262).setEnd(18627))
+            .setRangeDatetime(
+                TestRangeDatetime.newBuilder().setStart(1715360343).setEnd(1715446743))
+            .setRangeTimestamp(
+                TestRangeTimestamp.newBuilder().setStart(1715360343).setEnd(1715446743))
+            .setRangeDateMixedCase(TestRangeDate.newBuilder().setStart(1).setEnd(2))
+            .setRangeDatetimeMixedCase(
+                TestRangeDatetime.newBuilder()
+                    .setStart(142258614586538368L)
+                    .setEnd(142258525253402624L))
+            .setRangeTimestampMixedCase(
+                TestRangeTimestamp.newBuilder().setStart(10L).setEnd(1649174771000000L))
+            .build();
+
+    JSONArray data = new JSONArray();
+    JSONObject row = new JSONObject();
+
+    JSONObject rangeDate = new JSONObject();
+    rangeDate.put("start", 18262);
+    rangeDate.put("end", 18627);
+    row.put("range_date", rangeDate);
+
+    JSONObject rangeDatetime = new JSONObject();
+    rangeDatetime.put("start", 1715360343);
+    rangeDatetime.put("end", 1715446743);
+    row.put("range_datetime", rangeDatetime);
+
+    JSONObject rangeTimestamp = new JSONObject();
+    rangeTimestamp.put("start", 1715360343);
+    rangeTimestamp.put("end", 1715446743);
+    row.put("range_timestamp", rangeTimestamp);
+
+    JSONObject rangeDateMixedCase = new JSONObject();
+    rangeDateMixedCase.put("START", "1970-01-02");
+    rangeDateMixedCase.put("eND", "1970-01-03");
+    row.put("range_date_mixed_case", rangeDateMixedCase);
```

Comment:
The field name itself should be case insensitive.

Reply:
Updated.

Reply:
Where is the update? I mean to have name like this "Range_Date_Mixed_Case"